cuda/extensions.bzl

```diff
@@ -2,7 +2,28 @@
 
 load("//cuda/private:repositories.bzl", "local_cuda")
 
+cuda_toolkit = tag_class(attrs = {
+    "name": attr.string(doc = "Name for the toolchain repository", default = "local_cuda"),
+    "toolkit_path": attr.string(doc = "Path to the CUDA SDK"),
+})
+
 def _init(module_ctx):
-    local_cuda(name = "local_cuda")
+    registrations = {}
+    for mod in module_ctx.modules:
+        for toolchain in mod.tags.local_toolchain:
+            if not mod.is_root:
+                fail("Only the root module may override the path for the local cuda toolchain")
```

Comment:
Note that this is an explicit choice that a user has to define a cuda toolchain in their MODULE.bazel file. 
This is anticipating that we will be able to setup a fully hermetic toolchain and so the root MODULE will decide whether to use the locally installed cuda or the hermetic toolchain.